{{cookiecutter.project_slug}}/frontend/app/components/moderation/UserTable.tsx

```diff
@@ -0,0 +1,106 @@
+"use client"
+
+import { apiAuth } from "../../lib/api"
+import { IUserProfile } from "../../lib/interfaces"
+import CheckState from "./CheckState"
+import ToggleActive from "./ToggleActive"
+import ToggleMod from "./ToggleMod"
+import { useAppDispatch, useAppSelector } from "../../lib/hooks"
+import { RootState } from "../../lib/store"
+import { useEffect, useState } from "react"
+import { refreshTokens, token } from "../../lib/slices/tokensSlice"
+
+const renderUserProfiles = (userProfiles: IUserProfile[]) => {
+  return userProfiles.map((profile) => (
+    <tr key={profile.id}>
+      <td className="w-full max-w-0 py-4 pl-4 pr-3 text-sm font-medium text-gray-900 sm:w-auto sm:max-w-none sm:pl-6">
+        {profile.fullName}
+        <dl className="font-normal lg:hidden">
+          <dt className="sr-only">Email</dt>
+          <dd className="mt-1 truncate text-gray-700">{profile.email}</dd>
+          <dt className="sr-only sm:hidden">Validated</dt>
+          <dd className="mt-1 truncate sm:hidden">
+            <CheckState check={profile.email_validated} />
+          </dd>
+        </dl>
+      </td>
+      <td className="hidden px-3 py-4 text-sm text-gray-500 lg:table-cell">
+        {profile.email}
+      </td>
+      <td className="hidden px-3 py-4 text-sm lg:table-cell">
+        <CheckState check={profile.email_validated} />
+      </td>
+      <td className="px-3 py-4 text-sm text-gray-500">
+        <ToggleActive check={profile.is_active} email={profile.email} />
+      </td>
+      <td className="px-3 py-4 text-sm text-gray-500">
+        <ToggleMod check={profile.is_superuser} email={profile.email} />
+      </td>
+    </tr>
+  ))
+}
+
+export default function UserTable() {
+  const dispatch = useAppDispatch()
+  const accessToken = useAppSelector((state: RootState) => token(state))
+
+  const [userProfiles, setUserProfiles] = useState([] as IUserProfile[])
+
+  async function getAllUsers() {
+    await dispatch(refreshTokens())
+    const res = await apiAuth.getAllUsers(accessToken)
+    if (res && res.length) setUserProfiles(res)
+  }
+
+  useEffect(() => {
+    async function fetchUsers() {
+      await getAllUsers()
+    }
+
+    fetchUsers()
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Leaving a note here. I ran into an issue where the `getAllUsers` call constantly fired. I haven't checked it again yet, but I'll leave note just to come back and confirm this behavior is gone.